packages/ember-data/lib/system/store.js

```diff
@@ -584,21 +584,43 @@ Store = Service.extend({
     @method findRecord
     @param {String} modelName
     @param {(String|Integer)} id
-    @param {Object} preload - optional set of attributes and relationships passed in either as IDs or as actual models
+    @param {Object} options
     @return {Promise} promise
   */
-  findRecord: function(modelName, id, preload) {
+  findRecord: function(modelName, id, options) {
     Ember.assert('Passing classes to store methods has been removed. Please pass a dasherized string instead of '+ Ember.inspect(modelName), typeof modelName === 'string');
     var internalModel = this._internalModelForId(modelName, id);
 
-    return this._findByInternalModel(internalModel, preload);
+    // Detect preloading
+    if (options) {
+      var modelProperties = [];
+      var fields = Ember.get(internalModel.type, 'fields');
+      fields.forEach(function(fieldType, key) {
+        modelProperties.push(key);
+      });
+      var preloadDetected = modelProperties.reduce(function(memo, key) {
+        return typeof options[key] !== 'undefined' || memo;
+      }, false);
+      if (preloadDetected) {
```

Comment:
This should not be on `findRecord` as findRecord is a new api, we should add the deprecation only on `find` and `findById` if findById was public